pkg/bucketeer/api/rest.go

```diff
@@ -0,0 +1,228 @@
+	EvaluationEventType
+	MetricsEventType
+)
+
+const SourceIDGOSERVER = 5
+
+type successResponse struct {
+	Data json.RawMessage `json:"data"`
+}
+
+type registerEventsRequest struct {
+	Events []*Event `json:"events,omitempty"`
+}
+
+type RegisterEventsResponse struct {
+	Errors map[string]*RegisterEventsResponseError `json:"errors,omitempty"`
+}
+
+type getEvaluationRequest struct {
+	Tag       string     `json:"tag,omitempty"`
+	User      *user.User `json:"user,omitempty"`
+	FeatureID string     `json:"feature_id,omitempty"`
+	SourceID  int32      `json:"source_id,omitempty"`
+}
+
+type GetEvaluationResponse struct {
+	Evaluation *Evaluation `json:"evaluations,omitempty"`
+}
+
+type Event struct {
+	ID                   string          `json:"id,omitempty"`
+	Event                json.RawMessage `json:"event,omitempty"`
+	EnvironmentNamespace string          `json:"environment_namespace,omitempty"`
+	Type                 eventType       `json:"type,omitempty"`
+}
+
+type MetricsEvent struct {
+	Timestamp int64                  `json:"timestamp,omitempty"`
+	Event     json.RawMessage        `json:"event,omitempty"`
+	Type      metricsDetailEventType `json:"type,omitempty"`
+}
+
+type GoalEvent struct {
+	Timestamp int64      `json:"timestamp,omitempty"`
+	GoalID    string     `json:"goal_id,omitempty"`
+	UserID    string     `json:"user_id,omitempty"`
+	Value     float64    `json:"value,omitempty"`
+	User      *user.User `json:"user,omitempty"`
+	Tag       string     `json:"tag,omitempty"`
+	SourceID  int32      `json:"source_id,omitempty"`
+}
+
+type InternalErrorCountMetricsEvent struct {
+	Tag string `json:"tag,omitempty"`
+}
+
+type GetEvaluationSizeMetricsEvent struct {
+	Labels   map[string]string `json:"labels,omitempty"`
+	SizeByte int32             `json:"size_byte,omitempty"`
+}
+
+type GetEvaluationLatencyMetricsEvent struct {
+	Labels   map[string]string `json:"labels,omitempty"`
+	Duration time.Duration     `json:"duration,omitempty"`
+}
+
+type TimeoutErrorCountMetricsEvent struct {
+	Tag string `json:"tag,omitempty"`
+}
+
+type Variation struct {
+	ID    string `json:"id,omitempty"`
+	Value string `json:"value,omitempty"`
+	// number or even json object
+	Name        string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+}
+
+type Evaluation struct {
+	ID             string  `json:"id,omitempty"`
+	FeatureID      string  `json:"feature_id,omitempty"`
+	FeatureVersion int32   `json:"feature_version,omitempty"`
+	UserID         string  `json:"user_id,omitempty"`
+	VariationID    string  `json:"variation_id,omitempty"`
+	Reason         *Reason `json:"reason,omitempty"`
+	VariationValue string  `json:"variation_value,omitempty"`
+}
+
+type EvaluationEvent struct {
+	Timestamp      int64      `json:"timestamp,omitempty"`
+	FeatureID      string     `json:"feature_id,omitempty"`
+	FeatureVersion int32      `json:"feature_version,omitempty"`
+	UserID         string     `json:"user_id,omitempty"`
```

Comment:
Since we have the User ID in the User struct, I think we could remove it. What do you think?